pkg/querier/http.go

```diff
@@ -38,41 +40,53 @@ type QueryResponse struct {
 	Result     parser.Value     `json:"result"`
 }
 
-type Engine interface {
-	Query(logql.Params) logql.Query
-}
-
 // nolint // QuerierAPI defines HTTP handler functions for the querier.
 type QuerierAPI struct {
-	querier Querier
-	cfg     Config
-	limits  querier_limits.Limits
-	engine  Engine
+	querier  Querier
+	cfg      Config
+	limits   querier_limits.Limits
+	engineV1 logql.Engine // Loki's current query engine
+	engineV2 logql.Engine // Loki's next generation query engine
```

Comment:
Do you think it might be more self-doucmenting if these had the concrete types instead of the interface types?

```go 
engineV1 *logql.QueryEngine
engineV2 *engine.QueryEngine // Next-generation engine 
```

Reply:
This would defeat the purpose of the `Engine` interface, wouldn't it? 
I'm not completely opposed, though.

Reply:
I don't feel too strongly either way; my first question as a reader would be "ok, where is engineV1 and engineV2 implemented?" which takes slightly more effort when going to the constructor vs having them in the types to start with.

But it's such a tiny difference I don't think it matters much either way; feel free to leave it if you want.